File: frontend/src/api/listings.js
import client from "./client"

export const getListings    = (params) => client.get("/listings/", { params })
export const getListing     = (id)     => client.get(`/listings/${id}`)
export const getMyListings  = (params) => client.get("/listings/my/listings", { params })
export const createListing  = (data)   => client.post("/listings/", data)
export const updateListing  = (id, data) => client.patch(`/listings/${id}`, data)
export const cancelListing  = (id)    => client.delete(`/listings/${id}`)

File: backend/app/routers/listings.py
from fastapi import APIRouter
from pydantic import BaseModel

router = APIRouter()

class ListingCreate(BaseModel):
    title: str
    description: str
    starting_price: float
    auction_end_time: str

class ListingResponse(BaseModel):
    id: str
    title: str
    description: str
    starting_price: float
    current_price: float
    status: str

    class Config:
        from_attributes = True

@router.get("/")
async def get_listings():
    return {"message": "Listings endpoint - to be implemented"}

@router.get("/{listing_id}")
async def get_listing(listing_id: str):
    return {"message": f"Get listing {listing_id} - to be implemented"}

@router.post("/")
async def create_listing(listing: ListingCreate):
    return {"message": "Create listing - to be implemented"}

@router.put("/{listing_id}")
async def update_listing(listing_id: str, listing: ListingCreate):
    return {"message": f"Update listing {listing_id} - to be implemented"}

@router.delete("/{listing_id}")
async def delete_listing(listing_id: str):
    return {"message": f"Delete listing {listing_id} - to be implemented"}
